pkgs/applications/networking/instant-messengers/telegram/default.nix

```diff
@@ -0,0 +1,89 @@
+{ stdenv, fetchurl, fetchgit, xorg, glib, qt55, fontconfig, freetype, dbus
+, libxkbcommon, makeWrapper, libredirect, xkeyboard_config, makeDesktopItem }:
+
+stdenv.mkDerivation rec {
+  name = "telegram-${version}";
+
+  version = "0.9.10";
+
+  src = fetchurl {
+    urls = [
+      "https://updates.tdesktop.com/tlinux/tsetup.${version}.tar.xz"
```

Comment:
@jagajaga tdesktop.com reroutes to desktop.telegram.org, though I'd vote for the latter in the sources here.